magicl-transcendental.asd

```diff
@@ -2,9 +2,46 @@
 ;;;;
 ;;;; Author: Robert Smith
 
+;;; Adapted from commonqt's qt.asd
+(defclass f->so (asdf:source-file)
+  ()
+  (:default-initargs
+   :type "f"))
+
+(defmethod output-files ((operation compile-op) (component f->so))
+  (values (list (make-pathname :name "libexpokit"
+                               :type #-darwin "so" #+darwin "dylib"
+                               :defaults (component-pathname component)))
+          t))
+
+(defmethod perform ((operation load-op) (component f->so))
+  t)
+
+(defmethod perform ((operation compile-op) (component f->so))
+  (flet ((nn (x) (uiop:native-namestring x)))
+    (let* ((fortran-file (component-pathname component))
+           (object-file (make-pathname :type "o" :defaults fortran-file))
+           (shared-object (make-pathname :type #+darwin "dylib" #-darwin "so"
+                                         :name "libexpokit"
+                                         :defaults fortran-file)))
+      (uiop:run-program
+       (list "gfortran" "-fPIC" "-c"
+             (nn fortran-file)
+             "-o"
+             (nn object-file)))
+      (uiop:run-program
+       (list "gfortran" #+darwin "-dynamiclib" #-darwin "-shared"
+             "-o" (nn shared-object)
+             (nn object-file)
+             #+darwin "-lblas"
+             #+darwin "-llapack")))))
```

Comment:
do we need to remove the generated .o?

Reply:
I don't think it's required, but it would keep things tidy. I'll update.